website/release-notes-parser.ts

```diff
@@ -0,0 +1,52 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import fs from 'node:fs';
+
+export function parseNotes(filename: string, releaseVersion: string): void {
+  const folderName = './static/release-notes';
+  const fileContent = fs.readFileSync(filename, { encoding: 'utf-8' });
+  const resultText = fileContent.split('---', 3);
+
+  // get release image url
+  const imagePath = /image: (.+)\n/.exec(resultText[1]);
+  const imageName = imagePath ? imagePath[1] : '';
+  const imageUrl = imageName ? `https://podman-desktop.io${imageName}` : '';
+  const pageName = /slug: (.+)\n/.exec(resultText[1]);
+  const blogName = pageName ? pageName[1] : `podman-desktop-release-${releaseVersion}`;
```

Comment:
I think my comment may have been lost, but here the URL will depend on the slugName of the blog post, here we assume it'll be construct following a given scheme but it may change

shouldn't we look at the slug field from the markdown file ?

example with https://raw.githubusercontent.com/containers/podman-desktop/main/website/blog/2024-08-08-release-1.12.md
extract the `slug` being `slug: podman-desktop-release-1.12`

Reply:
@benoitf  L30 extracts slug, but if slug doesn't exists, it will use the given scheme. If we know for sure that slug is always given, I can remove L31